iphone/Classes/PlatformModule.m

```diff
@@ -295,17 +295,17 @@ - (NSNumber *)totalMemory
 }
 GETTER_IMPL(NSNumber *, availableMemory, AvailableMemory);
 
-- (BOOL)openURL:(NSString *)url withOptions:(id)options andCallback:(JSValue *)callback
+- (BOOL)openURL:(NSString *)url withOptions:(JSValue *)options andCallback:(JSValue *)callback
 {
   NSURL *newUrl = [TiUtils toURL:url proxy:self];
   BOOL result = NO;
 
-  // iOS 10+
   NSMutableDictionary *optionsDict = [NSMutableDictionary dictionary];
-  if ([options isKindOfClass:[NSDictionary class]]) {
-    optionsDict = (NSMutableDictionary *)options;
-  } else if ([options isKindOfClass:[JSValue class]]) {
-    callback = (JSValue *)options;
+  NSDictionary *dictionary = [options toDictionary];
```

Comment:
this call is ok if the second arg is a callback function?

Reply:
Yes. This will return blank dictionary.
But there is another issue in PR. I'll update PR.

Reply:
There is an 

> Can we write a quick unit test to verify the fix - basically ensure the callback gets fired when passing in no options?
> 
> ```js
> Ti.Platform.openURL('http://www.google.com', e => finish());
> ```
> 
> Or will it exit the app to Safari or something? Maybe there's a URL that can be handled by the test app itself or in a way that lets the test app keep going?

Yeah. For this particular issue a simple test case can be written. Open a custom url scheme which does not exist. e.g  
 Ti.Platform.openURL("randomapp://", e => finish());
I'll add this.